app/helpers/dropdown/letter_with_invoice_new.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2023, Die Mitte Schweiz. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
+module Dropdown
+  class LetterWithInvoiceNew < Base
+    def initialize(template, label: nil,
+                   path: template.path_args(Message::LetterWithInvoice),
+                   disabled_msg: nil, icon: :plus)
+      super(template, label, icon)
+      @label = label
+      @path = path
+      init_items
+    end
+
+    def button_or_dropdown
+      if InvoiceItem.all_types.one? || @disabled_msg.present?
+        single_button
+      else
+        to_s
+      end
+    end
+
+    private
+
+    def label
+      @label || I18n.t('crud.new.title', model: Message::LetterWithInvoice.model_name.human)
+    end
+
+    def single_button
+      return template.action_button(label, @path, @icon, disabled: @disabled_msg) if @disabled_msg
```

Comment:
I think the disabled message feature for this button might not be implemented.

![image](https://github.com/hitobito/hitobito/assets/655627/f41d9b9b-cc5e-481c-b886-2721f493608b)

I made the invoice_config of one layer invalid and was expecting the button to be disabled. Am I getting it wrong or is this something we have to take care of?

Reply:
maybe add also a spec that checks if button is disabled if the invoice_config is invalid